snap/info.go

```diff
@@ -1318,12 +1318,16 @@ func SortServices(apps []*AppInfo) (sorted []*AppInfo, err error) {
 
 	for _, app := range apps {
```

Comment:
I think it'd be useful to add a comment:
```
// identify the successors and predecessors of each app, input data set may be
// a subset of all apps in the snap (eg. when restarting only few select apps), thus
// make sure to look only at those after/before apps that are listed in the input
```

Reply:
ah I forgot to include this in my most recent push, I think I will add this in a followup unless the tests are unhappy and I need to change something else

Reply:
done in https://github.com/snapcore/snapd/pull/9894